File: 1/1000s/15s/1550/ThreeConsecutiveOdds.cpp
/*
https://leetcode.com/problems/three-consecutive-odds/description/?envType=daily-question&envId=2025-05-11
*/

class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        int sz=arr.size(), idx=0;
        while(true){
            if(idx+2>=sz) break;
            if((arr[idx]%2!=0)&&((arr[idx+1])%2!=0)&&((arr[idx+2])%2!=0)) return true;
            idx++;
        }
        return false;
    }
}; 

File: 1/1000s/15s/1550/ThreeConsecutiveOdds.py
# https://leetcode.com/problems/three-consecutive-odds/description/?envType=daily-question&envId=2025-05-11

class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:
        lengthOfArr=len(arr)
        beginIndex=0
        while(True):
            if beginIndex+2>=lengthOfArr:
                break
            if arr[beginIndex]%2!=0 and arr[beginIndex+1]%2!=0 and arr[beginIndex+2]%2!=0:
                return True
            beginIndex+=1
        return False